packages/studio-plugin/src/HmrManager.ts

```diff
@@ -1,78 +1,65 @@
 import ParsingOrchestrator from "./ParsingOrchestrator";
 import { StudioHMRPayload, StudioHMRUpdateID, UserPaths } from "./types";
-import { HmrContext, ViteDevServer } from "vite";
+import { ViteDevServer } from "vite";
 import VirtualModuleID from "./VirtualModuleID";
+import chokidar from "chokidar";
 import upath from "upath";
 
 /**
  * HmrManager is responsible for handling studio specific HMR updates.
  */
 export default class HmrManager {
   constructor(
+    private server: ViteDevServer,
     private orchestrator: ParsingOrchestrator,
-    private pathToUserProjectRoot: string,
     private userPaths: UserPaths
   ) {}
 
+  createWatcher(pathToUserProjectRoot: string) {
+    const watcher = chokidar.watch(upath.join(pathToUserProjectRoot, "src"));
+    watcher.on("change", this.handleHotUpdate);
+    watcher.on("unlink", this.handleHotUpdate);
```

Comment:
would we also want to watch for new files added?

Reply:
This wasn't something we had before and was hard to do before (https://github.com/yext/studio/pull/123) but with some tweaks I added it here! also added tests